Source/VMware.vSphereDSC/Tests/Unit/TestHelpers/VMware.VimAutomation.Core/VMware.VimAutomation.Core.psm1

```diff
@@ -507,7 +506,53 @@ Add-Type -TypeDefinition @"
             return this.Key.GetHashCode();
         }
     }
-}
+
+    public class SyslogConfig : System.IEquatable<SyslogConfig>
+    {
+        public SyslogConfig()
+        {
+        }
+
+        public string LogHost { get; set; }
+
+        public string LogDir { get; set; }
+
+        public bool LogDirUnique { get; set; }
+
+        public bool CheckSslCerts { get; set; }
+
+        public long DefaultRotate { get; set; }
+
+        public long DefaultSize { get; set; }
+
+        public long DefaultTimeout { get; set; }
+
+        public long QueueDropMark { get; set; }
+
+        public long DropLogRotate { get; set; }
+
+        public long DropLogSize { get; set; }
+
+        public bool Equals(SyslogConfig syslogConfig)
+        {
+            return syslogConfig != null && this.LogHost == syslogConfig.LogHost && this.CheckSslCerts == syslogConfig.CheckSslCerts &&
+                this.DefaultRotate == syslogConfig.DefaultRotate && this.DefaultSize == syslogConfig.DefaultSize &&
+                this.DefaultTimeout == syslogConfig.DefaultTimeout && this.DropLogRotate == syslogConfig.DropLogRotate &&
+                this.DropLogSize == syslogConfig.DropLogSize && this.LogDir == syslogConfig.LogDir &&
+                this.LogDirUnique == syslogConfig.LogDirUnique && this.QueueDropMark == syslogConfig.QueueDropMark;
+        }
+
+        public override bool Equals(object syslogConfig)
+        {
+            return this.Equals(syslogConfig as SyslogConfig);
+        }
+
+        public override int GetHashCode()
+        {
+            return this.LogHost.GetHashCode();
```

Comment:
Here you can add all other string and long properties of the class.

Reply:
Shall the properties be concatenated with an underscore?
Like you did here?
```powershell
return (this.Name + "_" + this.User).GetHashCode();
```

Reply:
Yes it looks like a valid expression.